framework/include/times/Times.h

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose includes
+#include "GeneralReporter.h"
+
+/**
+ * Times objects are under the hood Reporters, but limited to a vector of Real
+ */
+class Times : public GeneralReporter
+{
+public:
+  static InputParameters validParams();
+  Times(const InputParameters & parameters);
+  virtual ~Times() = default;
+
+  /// Getter for the full times vector
+  const std::vector<Real> & getTimes() const;
+
+  /// Getter for a set of the full times
+  const std::set<Real> getUniqueTimes() const
```

Comment:
It's not usual practice to return a `const` value, but it would be unusual for a user to want to modify this return value, so it's probably fine